Projektskabelon-master/src/main/java/FunctionLayer/OverlayMaterialCalculator.java

```diff
@@ -18,33 +18,33 @@
 
 
     //......................SPAER......................//
-    public static Material spaerOneWall(Wall wall) {
+    public static Material spaerOneWall(Wall wall) throws LoginSampleException {
         Material spaer = new Material();
         int quantity = OverlaySizeCalculator.spaerOnOneWall(wall);
         int size = OverlaySizeCalculator.spaerLengthOneWall(wall);
         spaer.setName("47X100 MM SPÆRTRÆ");
         spaer.setAmount(quantity);
         spaer.setSize(size);
         spaer.setComment("Horizontal framing");
-
+        spaer.setPrice(LogicFacade.getPrice(spaer.getId()));
```

Comment:
If you set the price here, it will be calculated for the needed size, not for the available one. I calculate de available sizes first in function layer